src/render/processor.rs

```diff
@@ -117,6 +117,12 @@ pub trait AudioProcessor: Send {
     fn onmessage(&mut self, msg: &mut dyn Any) {
         log::warn!("Ignoring incoming message");
     }
+
+    /// Method called before the AudioProcessor is recycled, i.e. when sent back
+    /// to the control thread when its rendering has finished.
+    fn release_resources(&mut self) {
```

Comment:
Yeah, this is really tricky..
If a user implements this interface incorrectly and its processor will be reused, very subtle bugs will arise...